From 83a7f0d4e41bb3ce128cdc9ef4622bd7231b2280 Mon Sep 17 00:00:00 2001
From: Milad Fa <mfarazma@redhat.com>
Date: Thu, 17 Mar 2022 12:35:51 -0400
Subject: [PATCH] S390: disable baseline compiler with compressed pointers

Will re-enable once a few issues have been resolved.

Change-Id: Ieed60efcb6d59cc5ca10931913dd4d9ffea8b9f1
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3532202
Reviewed-by: Victor Gomes <victorgomes@chromium.org>
Commit-Queue: Milad Farazmand <mfarazma@redhat.com>
Cr-Commit-Position: refs/heads/main@{#79533}
---
 src/flags/flag-definitions.h | 3 ++-
 test/mjsunit/mjsunit.status  | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/flags/flag-definitions.h b/src/flags/flag-definitions.h
index efbe9ec7f6c..6a084220a5e 100644
--- a/src/flags/flag-definitions.h
+++ b/src/flags/flag-definitions.h
@@ -195,7 +195,8 @@ struct MaybeBoolFlag {
 #define ENABLE_CONTROL_FLOW_INTEGRITY_BOOL false
 #endif
 
-#if V8_TARGET_ARCH_PPC || V8_TARGET_ARCH_PPC64
+#if V8_TARGET_ARCH_PPC || V8_TARGET_ARCH_PPC64 || \
+    (V8_TARGET_ARCH_S390X && COMPRESS_POINTERS_BOOL)
 // TODO(v8:11421): Enable Sparkplug for these architectures.
 #define ENABLE_SPARKPLUG false
 #else
diff --git a/test/mjsunit/mjsunit.status b/test/mjsunit/mjsunit.status
index 251b5ea21ca..c3592dbac1a 100644
--- a/test/mjsunit/mjsunit.status
+++ b/test/mjsunit/mjsunit.status
@@ -1556,7 +1556,7 @@
 
 ##############################################################################
 # TODO(v8:11421): Port baseline compiler to other architectures.
-['arch not in (x64, arm64, ia32, arm, mips64el, mipsel, riscv64, loong64, s390x)', {
+['arch not in (x64, arm64, ia32, arm, mips64el, mipsel, riscv64, loong64, s390x) or (arch == s390x and pointer_compression)', {
   'baseline/*': [SKIP],
   'regress/regress-1242306': [SKIP],
 }],
-- 
2.35.1

